lib/salus/scanners/report_go_dep.rb

```diff
@@ -7,38 +7,66 @@
 module Salus::Scanners
   class ReportGoDep < Base
     def run
-      unless should_run?
+      if @repository.go_mod_present?
+        record_dep_from_go_mod
+      elsif @repository.dep_lock_present?
+        record_dep_from_go_lock_package
+      else
         raise(
           InvalidScannerInvocationError,
-          'Cannot report on Go dependencies without a Gopkg.lock file.'
+          'Cannot report on Go dependencies without a Gopkg.lock or go.mod/go.sum file'
         )
       end
+    end
+
+    def record_dep_from_go_mod
+      Dir.chdir(@repository.path_to_repo.to_s) do
+        shell_return = run_shell("go list -json -m all")
```

Comment:
Do we know if go list accounts for transitive dependencies? I found this article here which provides some details on that https://dave.cheney.net/2014/09/14/go-list-your-swiss-army-knife but if it does that could be an easy win.